halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/appengine/AppengineAccountValidator.java

```diff
@@ -106,9 +107,14 @@ public void validate(ConfigProblemSetBuilder p, AppengineAccount account) {
     }
     
     try {
-      // Location is the only App Engine resource guaranteed to exist. The API only accepts '-' here
-      // rather than project name, because the list of locations is static and not a property of an individual project.
-      credentials.getAppengine().apps().locations().list("-").execute();
+      credentials.getAppengine().apps().get(project).execute();
+    } catch (GoogleJsonResponseException e) {
+      if (e.getStatusCode() == 404) {
+        p.addProblem(Severity.ERROR, "No appengine application found for project " + project + ".")
+         .setRemediation("Run `gcloud app create --region <region>` to create an appengine application.");
+      } else {
+        p.addProblem(Severity.ERROR, "Failed to connect to appengine Admin API: " + e.getMessage() + ".");
```

Comment:
You could just rethrow and have your outer exception handler provide you with the same exact message.